netlogo-gui/src/main/app/interfacetab/WidgetActions.scala

```diff
@@ -0,0 +1,138 @@
+package org.nlogo.app.interfacetab
+
+import java.awt.Rectangle
+import javax.swing.undo.AbstractUndoableEdit
+
+import org.nlogo.core.Widget
+
+object WidgetActions {
+
+  val undoManager = new org.nlogo.editor.UndoManager() {
+    // The default one doesn't work!
+    override def canRedo = editToBeRedone() != null
```

Comment:
the `canRedo` method of the UndoManager class is not working here. I tried to debug it, but was unable to get a solutions. So I mentioned it here in case someone wondered why the method has been overridden.